### ISSUE TITLE
Validation loss is incorrectly scaled when combined with gradient accumulation

### ISSUE THREAD

**Issue by jdinalt** [pytorch/torchtitan#1759]
### Bug description

When gradient accumulation is enabled, the validation loss is incorrect. This is a consequence of the loss_fn being shared between train and validation mode. The loss function is wrapped with rescale_accumulated_loss(), which rescales the loss by the number of gradient accumulation steps.

This works for train, as the loss from N accumulation steps is summed at the end of the accumulation, before being logged. In the case of validation, the loop assumes that each step is a normal step, and the reported loss is scaled by 1/N.

There are two separate pathways by which this occurs. In the normal case, the wrapped loss function is used "as-is." In the case of PP, the loss function is not directly used, but is still applied indirectly through the pipeline scheduler.

In both cases, the result is that the reported validation loss is divided by the number of accumulation steps.

### Versions

torch  2.10.0.dev20250915+cu126

Test configurations:
https://github.com/jdinalt/forgather/tree/main/examples/torchtitan/test_parallelisms

**Comment by jdinalt** [comment]
This demonstrates two simple configurations., where both are using 10 gradient accumulation steps. "Control" is with 1 GPU and "PP" is with two, configured for "1F1B." Both use a very simple Lllam3 model and the Tiny Stories dataset.

<img width="3637" height="1786" alt="Image" src="https://github.com/user-attachments/assets/e5cb1321-4c27-4eb0-9040-f173a1e1f19e" />

**Comment by jdinalt** [comment]
The proposed fix for the problem is in this PR:
https://github.com/pytorch/torchtitan/pull/1758
